chat-extensions-compose/src/main/kotlin/com/ably/chat/extensions/compose/Messages.kt

```diff
@@ -0,0 +1,137 @@
+package com.ably.chat.extensions.compose
+
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import com.ably.chat.Message
+import com.ably.chat.MessageEventType
+import com.ably.chat.MessagesSubscription
+import com.ably.chat.PaginatedResult
+import com.ably.chat.Room
+import com.ably.chat.RoomStatus.Attached
+import com.ably.chat.annotations.ExperimentalChatApi
+import com.ably.chat.discontinuityAsFlow
+import io.ably.lib.types.AblyException
+import io.ably.lib.types.ErrorInfo
+
+/**
+ * Fetches the next page of messages when the user scrolls close to the last visible message.
+ *
+ * @param scrollThreshold The number of items remaining before reaching the last item at which fetching the next page is triggered.
+ * @param limit The number of messages to load per page.
+ *
+ * @return A list of paginated messages.
+ */
+@ExperimentalChatApi
+@Composable
+@Suppress("LongMethod")
+public fun Room.collectAsPagingMessagesState(scrollThreshold: Int = 10, limit: Int = 100): PagingMessagesState {
```

Comment:
Maybe call this `fetchSize` or something like that? `limit` has implications that may not be clear I think

Reply:
Agree